include/alpaka/math/abs/AbsUniformCudaHipBuiltIn.hpp

```diff
@@ -30,32 +30,22 @@ namespace alpaka
         {
             //! The CUDA built in abs trait specialization.
             template<typename TArg>
-            struct Abs<AbsUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point<TArg>::value>>
+            struct Abs<AbsUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_signed_v<TArg>>>
             {
                 __device__ auto operator()(AbsUniformCudaHipBuiltIn const& abs_ctx, TArg const& arg)
                 {
                     alpaka::ignore_unused(abs_ctx);
-                    return ::abs(arg);
-                }
-            };
-            //! The CUDA built in abs double specialization.
-            template<>
-            struct Abs<AbsUniformCudaHipBuiltIn, double>
-            {
-                __device__ auto operator()(AbsUniformCudaHipBuiltIn const& abs_ctx, double const& arg)
-                {
-                    alpaka::ignore_unused(abs_ctx);
-                    return ::fabs(arg);
-                }
-            };
-            //! The CUDA built in abs float specialization.
-            template<>
-            struct Abs<AbsUniformCudaHipBuiltIn, float>
-            {
-                __device__ auto operator()(AbsUniformCudaHipBuiltIn const& abs_ctx, float const& arg) -> float
-                {
-                    alpaka::ignore_unused(abs_ctx);
-                    return ::fabsf(arg);
+
+                    if constexpr(std::is_same_v<TArg, float>)
+                        return ::fabsf(arg);
+                    else if constexpr(std::is_same_v<TArg, double>)
+                        return ::fabs(arg);
+                    else if constexpr(std::is_same_v<TArg, int>)
+                        return ::abs(arg);
+                    else if constexpr(std::is_same_v<TArg, long int>)
+                        return ::labs(arg);
+                    else
```

Comment:
IMO we should use `else if constexpr(std::is_same_v<TArg, long long int>)` and static assert at the end of the function to avoid any side effects, e.g. if `is_siogned` is defined for an data type but it is not `long long int`